src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -1,21 +1,20 @@
 package cromwell.engine.db
 
-import cromwell.engine.WorkflowOutputs
-import wdl4s._
 import cromwell.engine.ExecutionStatus.ExecutionStatus
-import cromwell.engine._
+import cromwell.engine.{WorkflowOutputs, _}
 import cromwell.engine.backend.{Backend, JobKey}
 import cromwell.engine.db.slick._
 import cromwell.engine.workflow.{BackendCallKey, ExecutionStoreKey, OutputKey}
 import cromwell.webservice.{CallCachingParameters, WorkflowQueryParameters, WorkflowQueryResponse}
+import wdl4s._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 object DataAccess {
   val globalDataAccess: DataAccess = new slick.SlickDataAccess()
 }
 
-trait DataAccess {
+trait DataAccess extends AutoCloseable {
```

Comment:
Yeah ARM & Scala has been a weird thing for a while now :)

As long as you're reasonably sure that it's doing what you think it's doing (or at least not doing anything unfriendly) I'm all for it - as I said, on paper it seems like a good idea. I was just a bit leery about it being fairly unvetted in the scala/slick context.